sandbox/modules/identity/src/main/java/org/opensearch/identity/rest/user/create/RestCreateUserAction.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.rest.user.create;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.identity.DefaultObjectMapper;
+import org.opensearch.identity.rest.IdentityRestConstants;
+import org.opensearch.identity.utils.ErrorType;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestChannel;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestStatusToXContentListener;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static org.opensearch.identity.utils.RoutesHelper.addRoutesPrefix;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+
+/**
+ * Rest action for creating a user
+ *
+ * /users/{user} rest request action to create a user
+ *
+ * @opensearch.api
+ */
+public class RestCreateUserAction extends BaseRestHandler {
+
+    @Override
+    public String getName() {
+        return IdentityRestConstants.IDENTITY_CREATE_USER_ACTION;
+    }
+
+    /**
+     * Rest request handler for creating a new user
+     * @param request the request to execute
+     * @param client  client for executing actions on the local node
+     * @return the action to be executed See {@link #handleRequest(RestRequest, RestChannel, NodeClient) for more}
+     * @throws IOException if errors encountered when parsing from XContent
+     */
+    @Override
+    public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        String username = request.param("name");
+
+        // Parsing request body using DefaultObjectMapper
+        JsonNode contentAsNode;
```

Comment:
Is there a  reason to pick the DefaultObjectMapper way versus just accessing the parameters?

Reply:
The request content comes in a ByteArray (even though it is a JSON). We pass this through DefaultObjectMapper to parse and create a JSON node, which we will then parse to extract the properties (`password` in our case)

Good question!